### PR TITLE
Format the project with 'dotnet format'

### PR DESCRIPTION
**Pull request by SecondNewtonLaw**
This Pull Request runs a simple `dotnet format`, my reason to apply this pr is to make the project formatting more uniform, in a way that makes contributing to it easier as the formatting is similar between the different source code files

### CODE REVIEW

**Review by csmir — COMMENTED** [review]
While this change is welcome, there are minor code issues that *may* surface because of this. It takes an obscenely long time to review all changes, so I will put this on hold until I find the time to do so.

**Comment by csmir** [comment]
To add, it may also be worthwhile to force format on merge actions, to continue this consistency into the future

**Comment by SecondNewtonLaw** [comment]
I agree that this kinds of changes are far too big, and they require time reading the code to make sure it works as it previously did, my main reason to this pr was that when previously I forked the project to contribute, I was taken a back by the formatting being messy, sometimes there was one formatting style and other times there was another, I just think it's for the better for this kinds of changes to happen, they make the code style more normalized throughout the project making it easier to work with.

I will wait until you have time to review the changes

File: src/Discord.Net.Analyzers/GuildAccessAnalyzer.cs
Comment (Misha-133) [inline src/Discord.Net.Analyzers/GuildAccessAnalyzer.cs]:
```suggestion
        private const string Title = "Limit command to Guild contexts.";
```
I think this shouldn't be changed

File: samples/InteractionFramework/Program.cs
Comment (Misha-133) [inline samples/InteractionFramework/Program.cs]:
```suggestion
            return false;
```

**Review by Misha-133 — CHANGES_REQUESTED** [review]

File: src/Discord.Net.Analyzers/GuildAccessAnalyzer.cs
Comment (SecondNewtonLaw) [reply src/Discord.Net.Analyzers/GuildAccessAnalyzer.cs]: I agree on this change, it changes the string, which shouldn't have happene.

File: samples/InteractionFramework/Program.cs
Comment (SecondNewtonLaw) [reply samples/InteractionFramework/Program.cs]: 👍

**Review by Misha-133 — APPROVED** [review]

**Comment by cjbonn** [comment]
The reformatting of MessageExtension, for example, seems arbitrary. It's fine as it currently is imo - is the formatting something that can be customized/configured for something the maintainers can all agree on? Is it possible for it to ignore dependency/import ordering, there's many changes with _just_ that.

**Comment by SecondNewtonLaw** [comment]
> The reformatting of MessageExtension, for example, seems arbitrary. It's fine as it currently is imo - is the formatting something that can be customized/configured for something the maintainers can all agree on? Is it possible for it to ignore dependency/import ordering, there's many changes with _just_ that. 

That is just part of `dotnet format` maybe it could be tweaked with an .editorconfig but I'm not sure, this pr is just a quick `dotnet format`, you can try and read documentation on the command to see if anything can be done for the reordering of the namespaces.

**Review by csmir — APPROVED** [review]
With the most recent changes, I dont have any further remarks. LGTM

**Comment by csmir** [comment]
Could you merge dev, rerun and apply the same changes we suggested, resolving the conflicts in this PR? I am willing to merge it afterwards

**Comment by SecondNewtonLaw** [comment]
> Could you merge dev, rerun and apply the same changes we suggested, resolving the conflicts in this PR? I am willing to merge it afterwards

Sure, I'll do it later

**Comment by SecondNewtonLaw** [comment]
> The reformatting of MessageExtension, for example, seems arbitrary. It's fine as it currently is imo - is the formatting something that can be customized/configured for something the maintainers can all agree on? Is it possible for it to ignore dependency/import ordering, there's many changes with _just_ that.

Regarding this, after looking at `dotnet format` help, you can set to ignore certain diagnostics
```
Options:
  --diagnostics <diagnostics>                                              A space separated list of diagnostic ids to use as a filter when fixing code style or 3rd party issues. []
  --exclude-diagnostics <exclude-diagnostics>                              A space separated list of diagnostic ids to ignore when fixing code style or 3rd party issues. []
  --severity <error|info|warn>                                             The severity of diagnostics to fix. Allowed values are info, warn, and error.
  --no-restore                                                             Doesn't execute an implicit restore before formatting.
  --verify-no-changes                                                      Verify no formatting changes would be performed. Terminates with a non-zero exit code if any files would have been formatted.
  --include <include>                                                      A list of relative file or folder paths to include in formatting. All files are formatted if empty. []
  --exclude <exclude>                                                      A list of relative file or folder paths to exclude from formatting. []
  --include-generated                                                      Format files generated by the SDK.
  -v, --verbosity <d|detailed|diag|diagnostic|m|minimal|n|normal|q|quiet>  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic]
  --binarylog <binary-log-path>                                            Log all project or solution load information to a binary log file.
  --report <report-path>                                                   Accepts a file path which if provided will produce a json report in the given directory.
  --version                                                                Show version information
  -?, -h, --help                                                           Show help and usage information
```

**Pull request merged**
